src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1369,12 +1369,42 @@ private BoundExpression BindNonMethod(SimpleNameSyntax node, Symbol symbol, Diag
                         }
                         else
                         {
+                            SourceLocalSymbol sourceLocal;
+                            type = null;
+
+                            if (node.SyntaxTree == localSymbolLocation.SourceTree && (object)(sourceLocal = localSymbol as SourceLocalSymbol) != null &&
+                                sourceLocal.IdentifierToken.Parent?.Kind() == SyntaxKind.Argument)
+                            {
+                                var argument = (ArgumentSyntax)sourceLocal.IdentifierToken.Parent;
+
+                                if (argument.Identifier == sourceLocal.IdentifierToken &&
+                                    argument.Type.IsVar)
+                                {
+                                    // We are referring to an out variable which might need type inference.
+                                    // If it is in fact needs inference, it is illegal to reference it in the same argument list. 
+                                    // Technically, we could support some restricted scenarios (when it is used as another argument in the same list), 
```

Comment:
I think you are confused. This place in code doesn't implement the constraint that you are suggesting to document. It is, however, implements a constraint that is documented in outvar.md on line 19
```
It is illegal to reference implicitly-typed out variable in the same argument list. 
Technically, we could support some restricted scenarios (when it is used as another argument in the same list), 
but their utility is very questionable. Otherwise, we are looking into getting into a cycle trying to infer its type.
```

---
In reply to: [64256897](https://github.com/dotnet/roslyn/pull/11493#discussion_r64256897) [](ancestors = 64256897)